src/os/bluestore/BlueStore.cc

```diff
@@ -8787,7 +8787,9 @@ void BlueStore::_kv_finalize_thread()
       }
 
       // this is as good a place as any ...
-      _reap_collections();
+      // _queue_reap_collection in this thread. So this judgement don't protected by reap_lock.
+      if (!removed_collections.empty())
```

Comment:
May be move inside reap_collections to have all the stuff in a single place?

Reply:
If move this inside reap_collection. We should protect by reap_lock. And there is only two caller: 1)this ;2)umount.  So i think this is better than move inside reap_collection.